docs/monorepo.md

```diff
@@ -133,22 +133,70 @@ Note that if you're building with Webpack, you may need to turn off [`resolve.sy
+We use [Lerna](https://lernajs.io/) and its `publish` command to publish the monorepo packages to NPM registry. Please do not use regular [`npm publish`](https://docs.npmjs.com/cli/publish) within a package to publish an individual package; `npx` has issues using this flow.
+
+### Make sure changelogs and `package.json` versions are up to date
+
+For all packages that you want to publish, make sure that their `package.json` versions are bumped. Decide carefully whether you want to publish a patch, a minor or a major update of the package. Be mindful about [semantic versioning](https://semver.org/).
+
+Make sure that the `CHANGELOG.md` document contains up-to-date information, with the `next` heading replaced with the version number that you are about to publish.
+
+Create PRs with the necessary changes and merge them to `master` before publishing. Lerna will add a `gitHead` field to each published package's `package.json`. That field contains the hash of the Git commit that the package was published from. It's better if this commit hash is a permanent one from the `master` branch, rather than an ephemeral commit from a local branch.
+
+### Checkout the latest master locally and build the packages
+
+Always publish from the latest `master` branch, so that the package contents come from a verified source that everyone has access to. It's too easy to publish a NPM package from a local branch, or even uncommitted local modifications that are invisible to anyone but you.
+
+Build the `dist/` directories (the transpiled package content that will be published) from scratch.
+
+```
+git checkout master
+git pull
+git status (should be clean!)
+npm run distclean
+npm ci
+npm run build-packages
+```
+
+### Publishing all outdated packages
+
+It might be good to start un-authenticated, since Lerna doesn't have `--dry-run` option like NPM does: `npm logout`.
+
+Now run: `npx lerna publish from-package`
+
+Lerna will show a list of packages that have versions higher than the latest one published in the NPM registry. Verify that this is indeed the list of packages that you want to publish. If something looks off, abort!
+
+If you say "yes" to the Lerna prompt, and are not authenticated, the publishing will fail with authentication error. Better to say "no".
+
+Now make sure you're logged in at this point, we're going to publish 🚀: `npm whoami`, `npm login`. Enter your username, password, and the OTP code.
+
+Craft the following command, we'll add `--yes` to skip prompts and save OTP cycle time. `--dist-tag next` is optional, use it when publishing unstable versions.
+
+Wait for your NPM OTP (one time password) cycle to start -- if your NPM account is correctly set up for maximum security, you'll need to enter the OTP code when publishing, even though you've already logged in with another OTP. Write the code into the command and publish:
+
+```
+NPM_CONFIG_OTP=[YOUR_OTP_CODE] npx lerna publish --dist-tag next from-package --yes
+```
```

Comment:
I think we can remove the OTP cycle time and `NPM_CONFIG_OTP=…` bits here. `lerna` now asks for OTP at the right time.

Reply:
Alright. I can confirm that Lerna/NPM asks interactively if you don't set the environment variable.